plugins/woocommerce/src/Internal/DataStores/Orders/DataSynchronizer.php

```diff
@@ -103,6 +110,7 @@ public function __construct() {
 		self::add_action( 'woocommerce_refund_created', array( $this, 'handle_updated_order' ), 100 );
 		self::add_action( 'woocommerce_update_order', array( $this, 'handle_updated_order' ), 100 );
 		self::add_action( 'wp_scheduled_auto_draft_delete', array( $this, 'delete_auto_draft_orders' ), 9 );
+		self::add_action( 'wp_scheduled_delete', array( $this, 'delete_trashed_orders' ) );
```

Comment:
WP hooks onto `wp_scheduled_delete` with the default priority of 10 (see https://github.com/WordPress/wordpress-develop/blob/5eef2bbb2a476fc3f94f7d7c2d89ff01b9b74f82/src/wp-includes/default-filters.php#L432). I believe it'd be best for us to hook onto this a bit earlier (priority 9, for example) so that we fully delete things thanks to our sync code. Otherwise, we could end up in a situation where WP deletes the post record _before_ our own code deletes the order from the HPOS tables.
This shouldn't be a problem per se, but I'm thinking it might actually simplify things.

Reply:
Yep, that does sound reasonable. ☺️ Addressed in [310ca5b](https://github.com/woocommerce/woocommerce/pull/41949/commits/310ca5ba502ea8234a6b38b1015494345d65328f).